api/src/main/java/org/openmrs/hl7/handler/ORUR01Handler.java

```diff
@@ -1293,7 +1293,7 @@ private void updateHealthCenter(Patient patient, PV1 pv1) {
 			
 			PersonAttribute currentHealthCenter = patient.getAttribute("Health Center");
 			
-			if (currentHealthCenter == null || !currentHealthCenter.equals(newLocationId.toString())) {
+			if (currentHealthCenter == null || !newLocationId.toString().equals(currentHealthCenter.getValue())) {
 				PersonAttribute newHealthCenter = new PersonAttribute(healthCenterAttrType, newLocationId.toString());
 				
 				log.debug("Updating patient's location from " + currentHealthCenter + " to " + newLocationId);

```

Comment:
@dkayiwa thanks. I tried with the above change. Still build fails. But now the exceptions are different. Earlier they were about FileNotFoundException. But now they are about mostly 
"Module unit test classes should extend BaseModuleContextSensitiveTest, not just BaseContextSensitiveTest"

full log : http://pastebin.com/pFAxdYWc
Do you have any ideas about them?